build-tools/xabootstrap/xabootstrap/ConfigAndData/Dependencies/AndroidToolchain.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Xamarin.Android.Bootstrap
+{
+	partial class AndroidToolchain : AppObject
+	{
+		public static readonly Uri AndroidUri = Configurables.Urls.AndroidToolchain_AndroidUri;
+		public static readonly Uri AntUri     = Configurables.Urls.AndroidToolchain_AntUri;
+
+		public List<AndroidToolchainComponent> Components { get; }
+
+		public AndroidToolchain ()
+		{
+			string AndroidNdkVersion       = BuildAndroidPlatforms.AndroidNdkVersion;
+			string AndroidNdkDirectory     = GetRequiredProperty (KnownProperties.AndroidNdkDirectory);
+			string AndroidCmakeVersion     = GetRequiredProperty (KnownProperties.AndroidCmakeVersion);
+			string AndroidCmakeVersionPath = GetRequiredProperty (KnownProperties.AndroidCmakeVersionPath);
+			string EmulatorVersion         = GetRequiredProperty (KnownProperties.EmulatorVersion);
+			string XABuildToolsFolder      = GetRequiredProperty (KnownProperties.XABuildToolsFolder);
+			string XABuildToolsVersion     = GetRequiredProperty (KnownProperties.XABuildToolsVersion);
+			string XAPlatformToolsVersion  = GetRequiredProperty (KnownProperties.XAPlatformToolsVersion);
+
+			Components = new List<AndroidToolchainComponent> {
+				new AndroidPlatformComponent ("android-2.3.3_r02-linux", "10"),
+				new AndroidPlatformComponent ("android-15_r03",          "15"),
+				new AndroidPlatformComponent ("android-16_r04",          "16"),
+				new AndroidPlatformComponent ("android-17_r02",          "17"),
+				new AndroidPlatformComponent ("android-18_r02",          "18"),
+				new AndroidPlatformComponent ("android-19_r03",          "19"),
+				new AndroidPlatformComponent ("android-20_r02",          "20"),
+				new AndroidPlatformComponent ("android-21_r02",          "21"),
+				new AndroidPlatformComponent ("android-22_r02",          "22"),
+				new AndroidPlatformComponent ("platform-23_r03",         "23"),
+				new AndroidPlatformComponent ("platform-24_r02",         "24"),
+				new AndroidPlatformComponent ("platform-25_r03",         "25"),
+				new AndroidPlatformComponent ("platform-26_r02",         "26"),
+				new AndroidPlatformComponent ("platform-27_r03",         "27"),
+				new AndroidPlatformComponent ("platform-28_r04",         "28"),
+
+				// arguments are: string name, string destDir, Uri relativeUrl = null, bool isMultiVersion = false, bool noSubdirectory = false
+				{ "docs-24_r01", "docs"},
```

Comment:
I think this would be easier to read & understand if the extension method *weren't* used here.  I'd never previously considered adding an `List<SpecificType>.Add(args...)` extension method before, and had to pause for a bit to understand what was going on.